sources/packages/web/src/components/common/LocationSummary.vue

```diff
@@ -31,14 +31,21 @@
         </div>
       </v-col>
       <v-col cols="1">
-        <v-btn
-          color="primary"
-          variant="text"
-          @click="$emit('editLocation', item.id)"
-          prepend-icon="fa:fa fa-gear"
+        <check-a-e-s-t-permission-role
+          :role="Role.InstitutionEditLocationDetails"
         >
-          Edit
-        </v-btn>
+          <template v-slot="{ isReadonly }">
+            <v-btn
+              :color="!isReadonly ? 'primary' : 'secondary'"
```

Comment:
Should we need to explicitly change the color here? Disable property will automatically handle the aesthetics right?

Reply:
next vuetify version as some feature, lets see, 
![image](https://user-images.githubusercontent.com/77353155/184256857-bc63481e-bd7b-44f8-887a-553e4f5fcc1a.png)